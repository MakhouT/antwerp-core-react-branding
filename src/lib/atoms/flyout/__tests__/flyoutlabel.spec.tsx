import * as React from "react";

import * as A from "../../../";
import { renderAndMatchSnapshot } from "../../../../__tests__/__fixtures__/testUtils";

describe("FlyoutLabel", () => {
    it("Renders default", () => {
        // Arrange
        const element = (<A.FlyoutLabel onClick={jest.fn()} />);

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders with child", () => {
        // Arrange
        const element = (<A.FlyoutLabel onClick={jest.fn()}><p>Child</p></A.FlyoutLabel>);

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders with children", () => {
        // Arrange
        const element = (<A.FlyoutLabel onClick={jest.fn()}>
            <p>Child 1</p>
            <p>Child 2</p>
        </A.FlyoutLabel>);

        // Act and assert
        renderAndMatchSnapshot(element);
    });
});
