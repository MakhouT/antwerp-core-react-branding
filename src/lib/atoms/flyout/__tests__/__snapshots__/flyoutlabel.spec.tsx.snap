// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlyoutLabel Renders default 1`] = `
<div
  className="m-flyout__label"
  onClick={[Function]}
/>
`;

exports[`FlyoutLabel Renders with child 1`] = `
<div
  className="m-flyout__label"
  onClick={[Function]}
>
  <p>
    Child
  </p>
</div>
`;

exports[`FlyoutLabel Renders with children 1`] = `
<div
  className="m-flyout__label"
  onClick={[Function]}
>
  <p>
    Child 1
  </p>
  <p>
    Child 2
  </p>
</div>
`;
